enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,1147 @@
+selection enables the AWS CCM to use [AWS Subnet Discovery](#aws-subnet-discovery) as it did previously with
+`platform.aws.subnets`. For ControlPlaneLBs and ClusterNodes, the subnets are selected based on whether they are
+public or private and whether the cluster is configured as internal or external. Refer to the table below for more
+details on auto-selection.
+
+Manual role selection is when the cluster admin assigns roles to the provided subnets, and the installer
+explicitly configures the subnets to be used according to the cluster admin's role selection. Every subnet must have
+a role, and the roles labeled as "Required in Manual Mode" in the table below must be assigned at least one subnet:
+
+| Role                   | Required in Manual Mode   | Auto-Selection Component | Auto-Selected Subnets Type[^1]                             |
+|------------------------|---------------------------|--------------------------|------------------------------------------------------------|
+| ClusterNode            | Yes                       | Installer                | Private[^2]                                                |
+| EdgeNode               | No                        | Installer                | Local Zone                                                 |
+| Bootstrap              | Yes                       | Installer                | Public or Private                                          |
+| IngressControllerLB    | Yes                       | AWS CCM                  | Public for External Cluster, Private for Internal Clusters |
+| ControlPlaneExternalLB | Yes (if external cluster) | Installer (CAPA)         | Public                                                     |
+| ControlPlaneInternalLB | Yes                       | Installer (CAPA)         | Private                                                    |
+[^1]: Describes what type of subnets will be selected for this role in automatic role selection.
+[^2]: Public for OPENSHIFT_INSTALL_AWS_PUBLIC_ONLY
+
+
+### Rejecting Untagged Subnets Proposal
+
+In BYO VPCs, untagged subnets are nuisance as the AWS CCM may select them, leading to various bugs, RFEs, and
+support cases. Since this proposal introduces a new subnet install-config field, it's a great opportunity
+to introduce a new behavior which will provide a better user experience. If `platform.aws.vpc.subnets`
+is using automatic role selection (i.e. subnet discovery), the installer will inspect the VPC for untagged
+subnets not already specified in `platform.aws.vpc.subnets` and reject the installation if any exist.
+
+It's important to note that while the installer will reject additional untagged subnets during installation,
+a cluster admin can still add untagged subnets after installation (Day 2), which may cause IngressControllers
+to automatically use these new subnets. This proposal does not attempt to prevent this type of misconfiguration.
```

Comment:
When considering the problems with installer validation of this, one thing that came to mind was whether the ingress controller could emit some sort of alert in this case. I don't know much about this topic, so not sure if it's viable/appropriate.